sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeServiceClientBuilder.java

```diff
@@ -103,14 +103,25 @@ public DataLakeServiceClientBuilder() {
         blobServiceClientBuilder.addPolicy(BuilderHelper.getBlobUserAgentModificationPolicy());
     }
 
+    private DataLakeServiceVersion getServiceVersion() {
+        return version != null ? version : DataLakeServiceVersion.getLatest();
+    }
+
+    private HttpPipeline constructPipeline() {
+        return (httpPipeline != null) ? httpPipeline : BuilderHelper.buildPipeline(
+            storageSharedKeyCredential, tokenCredential, azureSasCredential, endpoint, retryOptions, coreRetryOptions,
+            logOptions, clientOptions, httpClient, perCallPolicies, perRetryPolicies, configuration, audience, LOGGER);
+    }
+
     /**
      * @return a {@link DataLakeServiceClient} created from the configurations in this builder.
      * @throws IllegalStateException If multiple credentials have been specified.
      * @throws IllegalStateException If both {@link #retryOptions(RetryOptions)}
      * and {@link #retryOptions(RequestRetryOptions)} have been set.
      */
     public DataLakeServiceClient buildClient() {
-        return new DataLakeServiceClient(buildAsyncClient(), blobServiceClientBuilder.buildClient());
+        return new DataLakeServiceClient(buildAsyncClient(), blobServiceClientBuilder.buildClient(),
+            constructPipeline(), endpoint, getServiceVersion(), accountName, azureSasCredential, tokenCredential != null);
```

Comment:
This will end up creating two `HttpPipeline` if one isn't configured. This should have a slight change for now while upload/download operations still use sync-over-async:

```java
DataLakeServiceAsyncClient asyncClient = buildAsyncClient();
return new DataLakeServiceClient(asyncClient, blobServiceClientBuilder.buildClient(),
    asyncClient.getHttpPipeline(), endpoint, getServiceVersion(), accountName, azureSasCredential, tokenCredential != null);
```

Or, this can revert to the previous constructor parameters and the sync client can pull things out of the async client.

```java
DataLakeServiceClient(DataLakeServiceAsyncClient dataLakeServiceAsyncClient, BlobServiceClient blobServiceClient, String sasToken, boolean isTokenCredentialAuthenticated) {
    this.dataLakeServiceAsyncClient = dataLakeServiceAsyncClient;
    this.blobServiceClient = blobServiceClient;
    this.azureDataLakeStorage = new AzureDataLakeStorageRestAPIImplBuilder()
        .pipeline(asyncClient.getHttpPipeline())
        .url(asyncClient.getAccountUrl())
        .version(asyncClient.getServiceVersion().getVersion())
        .buildClient();
    this.serviceVersion = asyncClient.getServiceVersion();
    this.accountName = asyncClient.getAccountName();
    this.sasToken = sasToken;
    this.isTokenCredentialAuthenticated = isTokenCredentialAuthenticated;
}
```